src/effects/effectchain.cpp

```diff
@@ -151,37 +155,63 @@ void EffectChain::setEnabled(bool enabled) {
 }
 
 void EffectChain::enableForChannel(const ChannelHandleAndGroup& handle_group) {
-    if (!m_bAddedToEngine) {
+    // TODO(Be): remove m_enabledChannels from this class and move this logic
+    // to EffectChainSlot
+    bool bWasAlreadyEnabled = m_enabledChannels.contains(handle_group);
+    if (!bWasAlreadyEnabled) {
+        m_enabledChannels.insert(handle_group);
+    }
+
+    // The allocation of EffectStates below may be expensive, so avoid it if
+    // not needed.
+    if (!m_bAddedToEngine || bWasAlreadyEnabled) {
```

Comment:
Ok, it actually works. It is confusing that we can Enable channels for an effect chain. Isn't the effect routing an effect rack option? And can only be changed for effect chains loads to the Engine? 
But nevermind, this is probably nothing we should target here.

Reply:
Effect racks are a useless layer of complication that have no function and should be removed before any more work is done on the effects system. They do not maintain any state. Routing switches only exist for the EffectChain layer. The EffectRack subclasses for special functions (EQs, QuickEffects, Master EQ) should be EffectUnit subclasses.

Reply:
EffectRacks are needed because of quick effects and EQs. This ensures, that we have no blocked effects slot in the post fader effect racks. I think the issue here is that the m_enabledChannels is not a effect chain property like your comment describes. I think it is worth to fix it in this PR.

Reply:
The EQs and QuickEffects could be implemented as special EffectUnit subclasses similar to how they are special EffectRack subclasses now. There's no need for a layer above EffectUnits.

Reply:
This way we loose the existing co namespace and we need to reserve indexes for quick effects and EQs in the units. I do not like the idea. It works currently well and I have no fun to support more Co legacy aliases.

Reply:
It is not uncommon for people new to Mixxx who are developing a controller mapping to get confused by the useless EffectRack nomenclature. I think it would be helpful to remove this source of confusion. More importantly, it would make the effects system easier to maintain by removing unused complication.

Reply:
The EffectRack nomenclature is currently not useless. If you remove it you need to reintroduce an other layer of confusing. You have not purposed a replacement concept, so it is hard to argue. 
In any case, once a user has understand it it is quite clear, and if we change it we will confuse the user again.

Reply:
Please give some hints about your plans.This is important to find the right name fro the master EQ rack.

Reply:
When removing the effect rack layer
[EffectRack1_EffectUnitN_EffectM] would become [EffectUnitN_EffectM]
[EqualizerRack1_[ChannelN]_Effect1] would become [EqualizerEffectUnit_[ChannelN]_Effect1]
[QuickEffectRack1_[ChannelN]] would become [QuickEffectUnit_[ChannelN]]

The naming of the master EQ rack is unimportant though because it is only exposed through the preferences. It is not presented as something skins or controllers can interface with (technically they could, but no one would know that without reading the source code).

Reply:
This does Imho not simplify the situation in any way.
And when if it does, it breaks established interface.
We should really have a strong reason to do so.